### ISSUE TITLE
pixAutoPhotoinvert - the sequel

### ISSUE THREAD

**Issue by rmast** [DanBloomberg/leptonica#630]
![175789293-f39ddfdb-6f3e-4598-8d16-80a1f4a88b36](https://user-images.githubusercontent.com/3341558/179575323-234cebaf-bce1-41f3-ace1-9543b6e9e088.jpg)

results with defaults 0,NULL,NULL in

![Plaatje achteraf](https://user-images.githubusercontent.com/3341558/179575577-84d89298-3d36-4e40-9f9c-a1f0558381a6.png)

The KVK on top left should be left as it was, or at least the bottom half of the V should be one way or the other to be covered by MRC. The text print - wis - clear on the top right should invert.

The applied algorithm might not be sufficiënt. If so DjVu has an expired patented algorithm for foreground/background separation:
https://patents.google.com/patent/US6901169

However it performs less when there's noise in the scan that looks like holes in the mask:
https://github.com/jwilk/didjvu/issues/21

But in this example, straightly converted from an electronic PDF form, there should be no issue at all.

**Comment by DanBloomberg** [comment]
Thank you for pointing out the deficiency in the photoinvert algorithm!
This is pretty hard to make robust.
It did OK with the text in the larger block on the left, did not attempt to invert the text in the small block at UR, and made a mess of the "KVK" logo at UL.
I see that you've also noted deficiencies in the djvu algorithm which is about 20 years old.

I'll try to look at the leptonica failure on your example in the next week or so.

**Comment by rmast** [comment]
Yes, unfortunately DjVu-development of the segmenter has been stalled during the attempts to make it pay off by software patents which by now have expired. Don't know if those old algorithms are still worth something nowadays in the AI age.

Outlook voor Android downloaden<https://aka.ms/ghei36>
________________________________
